crates/polars-io/src/parquet/read/read_impl.rs

```diff
@@ -604,18 +604,15 @@ fn rg_to_dfs_par_over_rg(
             continue;
         }
 
-        row_groups.push((i, rg_md, rg_slice, row_count_start));
+        row_groups.push((rg_md, rg_slice, row_count_start));
     }
 
     let dfs = POOL.install(|| {
         // Set partitioned fields to prevent quadratic behavior.
         // Ensure all row groups are partitioned.
         row_groups
             .into_par_iter()
-            .enumerate()
-            .map(|(iter_idx, (_rg_idx, _md, slice, row_count_start))| {
-                let md = &file_metadata.row_groups[iter_idx];
-
+            .map(|(md, slice, row_count_start)| {
```

Comment:
`iter_idx` was an index into the previous version of partitioned metadata - I was just meant to restore this to directly use `md` from `par_iter()`